BuildMasterAutomation/Functions/Get-BMDeployment.ps1

```diff
@@ -30,9 +50,35 @@ function Get-BMDeployment
         [Object] $Session,
 
         # The deployment to get. You can pass a deployment id or object.
-        [Parameter(Mandatory)]
         [Alias('ID')]
-        [Object] $Deployment
+        [Object] $Deployment,

```

Comment:
Put this parameter into a `ByID` parameter set on its own. If we pass a deployment ID we should get just that one deployment. Put the rest of the parameters into a `ByFilter` parameter set.